tests/testlib/s2n_ktls_test_utils.c

```diff
@@ -56,22 +56,21 @@ ssize_t s2n_test_ktls_sendmsg_io_stuffer(void *io_context, const struct msghdr *
 
     size_t total_len = 0;
     for (size_t count = 0; count < msg->msg_iovlen; count++) {
+        POSIX_ENSURE_REF(msg->msg_iov);
         uint8_t *buf = msg->msg_iov[count].iov_base;
-        POSIX_ENSURE_REF(buf);
         size_t len = msg->msg_iov[count].iov_len;
 
-        if (s2n_stuffer_write_bytes(&io_ctx->data_buffer, buf, len) < 0) {
-            /* This mock implementation only handles partial writes for msg_iovlen == 1.
-             *
-             * This simplifies the implementation and importantly doesn't limit our test
-             * coverage because partial writes are handled the same regardless of
-             * msg_iovlen. */
-            POSIX_ENSURE(msg->msg_iovlen == 1, S2N_ERR_SAFETY);
-
+        if (s2n_stuffer_write_bytes(data_buffer, buf, len) != S2N_SUCCESS) {
+            size_t partial_len = MIN(len, s2n_stuffer_space_remaining(data_buffer));
+            if (s2n_stuffer_write_bytes(data_buffer, buf, partial_len) == S2N_SUCCESS) {
+                total_len += partial_len;
+            }
```

Comment:
Wouldn't this always succeed?

Reply:
Anything that that can return an error will ;) But looking at this, an error here should actually be fatal, not EAGAIN. I changed it to just use POSIX_GUARD.